packages/jellyfish-api-core/__tests__/category/loan/takeLoan.test.ts

```diff
@@ -0,0 +1,244 @@
+    await tGroup.waitForSync()
+
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '30@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Loan cannot be taken on token with id (2) as "mintable" is currently false')
+
+    await tGroup.get(0).container.call('updateloantoken', ['TSLA', { mintable: true }])
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+  })
+
+  it('should takeLoan', async () => {
+    const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
+    expect(vaultBefore.loanSchemeId).toStrictEqual('scheme')
+    expect(vaultBefore.ownerAddress).toStrictEqual(vaultAddress)
+    expect(vaultBefore.isUnderLiquidation).toStrictEqual(false)
+    expect(vaultBefore.collateralAmounts).toStrictEqual(['10000.00000000@DFI', '1.00000000@BTC'])
+    expect(vaultBefore.collateralValue).toStrictEqual(15000)
+    expect(vaultBefore.loanAmount).toStrictEqual([])
+    expect(vaultBefore.loanValue).toStrictEqual(0)
+    expect(vaultBefore.currentRatio).toStrictEqual(-1) // empty loan
+
+    const vaultBeforeTSLAAcc = vaultBefore.loanAmount.length > 0
+      ? vaultBefore.loanAmount.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const vaultBeforeTSLAAmt = vaultBeforeTSLAAcc !== undefined ? Number(vaultBeforeTSLAAcc.split('@')[0]) : 0
+
+    const txid = await tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '40@TSLA'
+    })
+    expect(typeof txid).toStrictEqual('string')
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    const vaultAfter = await tGroup.get(0).container.call('getvault', [vaultId])
+    expect(vaultAfter.loanSchemeId).toStrictEqual(vaultBefore.loanSchemeId)
+    expect(vaultAfter.ownerAddress).toStrictEqual(vaultBefore.ownerAddress)
+    expect(vaultAfter.isUnderLiquidation).toStrictEqual(vaultBefore.isUnderLiquidation)
+    expect(vaultAfter.collateralAmounts).toStrictEqual(vaultBefore.collateralAmounts)
+    expect(vaultAfter.collateralValue).toStrictEqual(vaultBefore.collateralValue)
+
+    const interestInfo = await tGroup.get(0).container.call('getinterest', ['scheme', 'TSLA'])
+    const loanAmount = new BigNumber(40).plus(new BigNumber(40).multipliedBy(interestInfo[0].totalInterest))
+    expect(vaultAfter.loanAmount).toStrictEqual([loanAmount.toFixed(8) + '@TSLA']) // 40.00002280@TSLA
+    expect(vaultAfter.loanValue).toStrictEqual(loanAmount.multipliedBy(2).toNumber())
+    expect(vaultAfter.currentRatio).toStrictEqual(Math.round(vaultAfter.collateralValue / vaultAfter.loanValue * 100))
+
+    const vaultAfterTSLAAcc = vaultAfter.loanAmount.find((amt: string) => amt.split('@')[1] === 'TSLA')
+    const vaultAfterTSLAAmt = Number(vaultAfterTSLAAcc.split('@')[0])
+    expect(vaultAfterTSLAAmt - vaultBeforeTSLAAmt).toStrictEqual(40.00002280)
+  })
+
+  it('should takeLoan with utxos', async () => {
```

Comment:
I got this when I run this test separately

<img width="988" alt="Screenshot 2021-09-23 at 1 17 22 PM" src="https://user-images.githubusercontent.com/7101806/134457805-698db313-6ef6-4000-b8f0-ea1468f3a09c.png">